go-controller/pkg/ovn/master.go

```diff
@@ -1205,9 +1203,24 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 		if !ok {
 			return fmt.Errorf("spurious object in syncNodes: %v", tmp)
 		}
-		foundNodes.Insert(node.Name)
-
 		hostSubnets, _ := util.ParseNodeHostSubnetAnnotation(node)
+		if config.HybridOverlay.Enabled && len(hostSubnets) == 0 && houtil.IsHybridOverlayNode(node) {
+			// this is a hybrid overlay node so mark as allocated from the hybrid overlay subnet allocator
+			hostSubnet, err := houtil.ParseHybridOverlayHostSubnet(node)
+			if err != nil {
+				klog.Warning(err.Error())
+			} else if hostSubnet != nil {
+				klog.V(5).Infof("Node %s contains subnets: %v", node.Name, hostSubnet)
+				if err := oc.hybridOverlaySubnetAllocator.MarkAllocatedNetwork(hostSubnet); err != nil {
+					utilruntime.HandleError(err)
+				}
+			}
+			// there is nothing left to be done if this is a hybrid overlay node
+			continue
+
```

Comment:
@JacobTanenbaum remove the added newline?

Reply:
done